### ISSUE TITLE
Issue with PEFT model save_pretrained

### ISSUE THREAD

**Issue by Quicksilver466** [huggingface/peft#2001]
### System Info

Model: Custom Phi-3 implementation, but also causes issue with normal Phi-3
OS: Linux(ubuntu)
Transformers version: 4.43.3
PEFT version: 0.11.1

### Information

- [ ] The official example scripts
- [X] My own modified scripts

### Tasks

- [ ] An officially supported task in the `examples` folder
- [X] My own task or dataset (give details below)

### Reproduction

```python
from peft import LoraConfig, get_peft_model, prepare_model_for_kbit_training, AutoPeftModelForCausalLM, PeftModel, PeftConfig, PeftModelForCausalLM

from transformers import AutoModelForCausalLM

model_path = "/content/gdrive/My Drive/Colab Notebooks/DOE/src/models/weights/Phi-3-mini-128k-instruct"

model = AutoModelForCausalLM.from_pretrained(model_path)

peft_config = LoraConfig(**{
    "r": 16,
    "lora_alpha": 32,
    "lora_dropout": 0.05,
    "bias": "none",
    "task_type": "CAUSAL_LM",
    "target_modules": "all-linear",
    "modules_to_save": None,
    "base_model_name_or_path": "/content/gdrive/My Drive/Colab Notebooks/DOE/src/models/weights/Phi-3ex"
})

kbit_model = prepare_model_for_kbit_training(model)
peft_model = get_peft_model(kbit_model, peft_config)
peft_model.save_pretrained("/content/gdrive/MyDrive/Colab Notebooks/DOE/test_peft_model/phi-peft-v3")

``` 

### Expected behavior

The resultant adapter_config.json is following JSON:
```
{
  "alpha_pattern": {},
  "auto_mapping": null,
  "base_model_name_or_path": "/content/gdrive/My Drive/Colab Notebooks/DOE/src/models/weights/Phi-3-mini-128k-instruct",
  "bias": "none",
  "fan_in_fan_out": false,
  "inference_mode": true,
  "init_lora_weights": true,
  "layer_replication": null,
  "layers_pattern": null,
  "layers_to_transform": null,
  "loftq_config": {},
  "lora_alpha": 32,
  "lora_dropout": 0.05,
  "megatron_config": null,
  "megatron_core": "megatron.core",
  "modules_to_save": null,
  "peft_type": "LORA",
  "r": 16,
  "rank_pattern": {},
  "revision": null,
  "target_modules": [
    "gate_up_proj",
    "o_proj",
    "qkv_proj",
    "dAnd these can be still setup in private environmentown_proj"
  ],
  "task_type": "CAUSAL_LM",
  "use_dora": false,
  "use_rslora": false
}
``` 

But I would ideally want  _**"/content/gdrive/My Drive/Colab Notebooks/DOE/src/models/weights/Phi-3ex"**_ to be the value for _**base_model_name_or_path**_ in resultant adapter_config.

In my use case I am using a custom version of Phi-3 model(I have subclassed Phi-3 and overridden some methods and changed the architecture a bit), so when I create the peft_model by passing my modified model, the adapter_config sets the value of **_base_model_name_or_path_** to  _**Phi-3-mini-128k-instruct**_. Obviously when loading this peft model with `from_pretrained`, it can't find the base model and results in an error. Thus I would appreciate if I can know where I might be going wrong or if this an actual bug.

**Comment by BenjaminBossan** [comment]
> Obviously when loading this peft model with `from_pretrained`, it can't find the base model and results in an error.

How do you load the model? Should this not work?

```python
model_path = "/content/gdrive/My Drive/Colab Notebooks/DOE/src/models/weights/Phi-3-mini-128k-instruct"
model = AutoModelForCausalLM.from_pretrained(model_path)
peft_model = PeftModel.from_pretrained(model, "/content/gdrive/MyDrive/Colab Notebooks/DOE/test_peft_model/phi-peft-v3")
```

**Comment by Quicksilver466** [comment]
Yup its working, completely forgot we could do that. Sorry for the trouble. I would also look for why the way I was doing wasn't working i.e with specifying the key to base model path, also please let me know if you know.

**Comment by BenjaminBossan** [comment]
> would also look for why the way I was doing wasn't working

In this line, `base_model_name_or_path` is overridden:

https://github.com/huggingface/peft/blob/41c274ecac6247a63e3b10f8536904f3ead82213/src/peft/mapping.py#L166

> also please let me know the reason, if you are aware of it

This was added a long time ago by Younes, who no longer works with HF. Also, it was [committed directly](https://github.com/huggingface/peft/commit/6c9534e660de73fa44abaefbf1f4f5b0df818fbb), no PR, so there is no reason mentioned for this change. If I had to guess, this change makes it easier for users to see what the underlying base model was and it should also facilitate discovery on HF Hub.

But regardless of the actual reasons, I don't think we can change it anymore, as that would break backwards compatibility. I could see adding a warning to the quoted line if the model names differ from what is set in the peft config, not sure if that would have helped you in this situation.
